src/app/components/crypto-asset-item/crypto-asset-item.layout.tsx

```diff
@@ -33,30 +37,37 @@ export function CryptoAssetItemLayout({
   onSelectAsset,
   titleLeft,
   titleRightBulletInfo,
+  renderRightElement,
 }: CryptoAssetItemLayoutProps) {
-  const { availableBalanceString, dataTestId, formattedBalance } =
-    parseCryptoAssetBalance(availableBalance);
+  const symbol = availableBalance?.symbol ?? captionLeft;
+  const dataTestId = getCryptoDataTestId(symbol);
+
+  const { availableBalanceString, formattedBalance } = availableBalance
+    ? parseCryptoAssetBalance(availableBalance)
+    : { availableBalanceString: undefined, formattedBalance: undefined };
 
-  const titleRight = (
+  const titleRight: ReactNode | undefined = availableBalance ? (
```

Comment:
@cbadawi can you please refactor this to not use nested ternary statements? 

Nesting ternary statements makes the code hard to read and the logic tricky to follow.

I'm also unsure about adding all of this code for optional props based on `availableBalance` or not